### PR TITLE
Issue 25268 update local file config

### PR DESCRIPTION
**Pull request by rewixe**
## Hey, I just made a Pull Request!

The documentation in this page which gives examples of [local file configurations](https://backstage.io/docs/features/software-catalog/configuration/#local-file-type-file-configurations) was inaccurate and missing details, for example the catalog examples link did not point to anything, and it also said you are also not able to use placeholders in them like $text whereas like text would be yaml, json, etc. Adding this information to this doc
image

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

resolves https://github.com/backstage/backstage/issues/25268

### CODE REVIEW

File: docs/features/software-catalog/configuration.md [hunk 1]
```diff
@@ -47,7 +47,7 @@ action is logged for further investigation.
 ### Local File (`type: file`) Configurations
 
 In addition to url locations, you can use the `file` location type to bring in content from the local file system. You should only use this for local development, test setups, and example data, not for production data.
-You are also not able to use placeholders in them like `$text`. You can however reference other files relative to the current file. See the full [catalog example data set here](https://github.com/backstage/backstage/tree/master/packages/catalog-model/examples) for an extensive example.
+You are also not able to use placeholders in them like `$text`, `$json` or `$yaml`. You can however reference other files relative to the current file. See the full [catalog example data set here](https://github.com/backstage/backstage/blob/master/packages/create-app/templates/default-app/app-config.yaml.hbs) for an extensive example.
```
Comment (awanlin) [inline docs/features/software-catalog/configuration.md:50]: This link change is not correct, if you go to the current link you can clearly see the `all.yaml` file mentioned further on (line 52). There are numerous examples within this folder and its sub-folders.

**Review by awanlin — CHANGES_REQUESTED** [review]

File: docs/features/software-catalog/configuration.md (hunk 1 above)
Comment (rewixe) [reply docs/features/software-catalog/configuration.md:50]: Ok sure, I will change to the original link

**Review by awanlin — APPROVED** [review]

**Comment by awanlin** [comment]
Thanks @rewixe 🚀 

@camilaibs this is good to go, just documentation changes 👍

**Comment by awanlin** [comment]
Hi @Rugvip, this is just docs and good to go, can you approve, please?

**Review by Rugvip — APPROVED** [review]
Yep, thank you! 👍

**Pull request merged**